### PR TITLE
Product list bulk delete

### PR DESCRIPTION
**Pull request by poulch**
I want to merge this change because it fixes https://github.com/saleor/saleor-dashboard/issues/3540

**PR intended to be tested with API branch:** 

### Pull Request Checklist

1. [x] This code contains UI changes
2. [x] All visible strings are translated with proper context including data-formatting
3. [x] Attributes `[data-test-id]` are added for new elements
4. [x] Changes are mentioned in the changelog
5. [x] The changes are tested in different browsers and in light/dark mode

### Test environment config

API_URI=https://automation-dashboard.staging.saleor.cloud/graphql/
APPS_MARKETPLACE_API_URI=https://apps.staging.saleor.io/api/v2/saleor-apps

### Do you want to run more stable tests?

To run all tests, just select the stable checkbox. To speed up tests, increase the number of containers. Tests will be re-run only when the "run e2e" label is added.

1. [x] stable
2. [ ] giftCard
3. [ ] category
4. [ ] collection
5. [ ] attribute
6. [ ] productType
7. [ ] shipping
8. [ ] customer
9. [ ] permissions
10. [ ] menuNavigation
11. [ ] pages
12. [ ] sales
13. [ ] vouchers
14. [ ] homePage
15. [ ] login
16. [ ] orders
17. [ ] products
18. [ ] app
19. [ ] plugins
20. [ ] translations
21. [ ] navigation
22. [ ] variants
23. [ ] payments

CONTAINERS=6

### CODE REVIEW

File: src/products/views/ProductList/ProductList.tsx [hunk 1]
```diff
@@ -223,14 +232,30 @@ export const ProductList: React.FC<ProductListProps> = ({ params }) => {
...
-        reset();
+        clearRowSelection();
       }
     },
   });
 
+  const [productBulkDelete, productBulkDeleteOpts] =
+    useProductBulkDeleteMutation({
+      onCompleted: data => {
+        if (data.productBulkDelete.errors.length === 0) {
```
Comment (krzysztofzuraw) [inline src/products/views/ProductList/ProductList.tsx:243]: **question**: do we have error handling? or we are covered by global error handling?

**Review by krzysztofzuraw — COMMENTED** [review]
Should the `Delete` button cover the first row?
CleanShot 2023-04-26 at 15 23 46@2x

**Review by Droniu — APPROVED** [review]

File: src/products/views/ProductList/ProductList.tsx (hunk 1 above)
Comment (poulch) [reply src/products/views/ProductList/ProductList.tsx:244]: Yes it will be covered globally

**Review by krzysztofzuraw — APPROVED** [review]

**Review by krzysztofzuraw — APPROVED** [review]

**Review by andrzejewsky — CHANGES_REQUESTED** [review]
checkboxes in rows are not clickable

**Review by andrzejewsky — APPROVED** [review]

**Review by krzysztofzuraw — DISMISSED** [review]

**Comment by wojteknowacki** [comment]
Delete button present when the product is no longer checked
https://user-images.githubusercontent.com/124166231/236763481-fa4c9b63-ee8b-497b-917d-9aae827e650b.mov

**Comment by wojteknowacki** [comment]
When user did not decide to delete product 
Then checkbox is unchecked automatically 
And tooltip is still present

https://user-images.githubusercontent.com/124166231/236764404-ef7c32e6-f368-4f11-946b-371ee7d529d8.mov

**Comment by wojteknowacki** [comment]
Should adding new column remove row checks?

https://user-images.githubusercontent.com/124166231/236771188-d70a4e85-1e05-4b2a-9017-acd3eba64372.mov

**Comment by wojteknowacki** [comment]
Product column is not movable 


https://user-images.githubusercontent.com/124166231/236773209-27c89f22-3c04-42ca-b7d7-cd246a7fa606.mov

**Review by krzysztofzuraw — APPROVED** [review]

**Comment by wojteknowacki** [comment]
When deleting row eg 2nd "check" state remains and marks product which is now on 2nd position - same when it comes to pagination - check state remains when triggers pagination

https://github.com/saleor/saleor-dashboard/assets/124166231/8e3b8ee7-6fa5-4a4b-883c-e613fd97fc84


https://github.com/saleor/saleor-dashboard/assets/124166231/6ad7d18f-609b-455e-9bce-926b94e20fef

File: src/components/Datagrid/Datagrid.tsx [hunk 2]
```diff
@@ -151,6 +157,16 @@ export const Datagrid: React.FC<DatagridProps> = ({
   const [selection, setSelection] = useState<GridSelection>();
   const [hoverRow, setHoverRow] = useState<number | undefined>(undefined);
 
+  // Allow to listen to which  row is selected and notfiy parent component
```
Comment (Droniu) [inline src/components/Datagrid/Datagrid.tsx:160]: **nitpick**
```suggestion
  // Allow to listen to which  row is selected and notfiy parent component
```

Comment (Droniu) [inline src/components/Datagrid/Datagrid.tsx:160]: **nitpick**
```suggestion
  // Allow to listen to which row is selected and notfiy parent component
```

File: src/products/components/ProductListDeleteButton/ProductListDeleteButton.tsx [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+  return (
+    <Tooltip>
+      <Tooltip.Trigger>
+        <Button
+          ref={ref}
+          onClick={onClick}
+          icon={<TrashBinIcon />}
+          variant="secondary"
+        ></Button>
```
Comment (Droniu) [inline src/products/components/ProductListDeleteButton/ProductListDeleteButton.tsx:26]: **nitpick**
```suggestion
        />
```

File: src/products/views/ProductList/ProductList.tsx [hunk 4]
```diff
@@ -470,23 +500,31 @@ export const ProductList: React.FC<ProductListProps> = ({ params }) => {
...
-        onClose={closeModal}
-        onConfirm={() => {
-          productBulkDelete({
-            variables: { ids: listElements },
-          });
+        onClose={() => {
+          closeModal();
+          setTimeout(() => {
+            deleteButtonRef.current.blur();
+          }, 0);
```
Comment (Droniu) [inline src/products/views/ProductList/ProductList.tsx:525]: **question:** Why is setTimeout used here?

**Review by Droniu — APPROVED** [review]

**Review by Droniu — APPROVED** [review]

**Review by andrzejewsky — APPROVED** [review]

**Review by wojteknowacki — APPROVED** [review]

**Pull request merged**